text/src/autogluon/text/automm/models/ft_transformer.py

```diff
@@ -0,0 +1,596 @@
+            nn.init.uniform_(x, a=-d_sqrt_inv, b=d_sqrt_inv)
+        elif self == _TokenInitialization.NORMAL:
+            nn.init.normal_(x, std=d_sqrt_inv)
+
+
+class MultiheadAttention(nn.Module):
+    """Multihead Attention (self-/cross-) with optional 'linear' attention.
+
+    To learn more about Multihead Attention, see [devlin2018bert]. See the implementation
+    of `Transformer` and the examples below to learn how to use the compression technique
+    from [wang2020linformer] to speed up the module when the number of tokens is large.
+
+    Examples:
+        .. testcode::
+
+            n_objects, n_tokens, d_token = 2, 3, 12
+            n_heads = 6
+            a = torch.randn(n_objects, n_tokens, d_token)
+            b = torch.randn(n_objects, n_tokens * 2, d_token)
+            module = MultiheadAttention(
+                d_token=d_token, n_heads=n_heads, dropout=0.2, bias=True, initialization='kaiming'
+            )
+
+            # self-attention
+            x, attention_stats = module(a, a, None, None)
+            assert x.shape == a.shape
+            assert attention_stats['attention_probs'].shape == (n_objects * n_heads, n_tokens, n_tokens)
+            assert attention_stats['attention_logits'].shape == (n_objects * n_heads, n_tokens, n_tokens)
+
+            # cross-attention
+            assert module(a, b, None, None)
+
+            # Linformer self-attention with the 'headwise' sharing policy
+            k_compression = torch.nn.Linear(n_tokens, n_tokens // 4)
+            v_compression = torch.nn.Linear(n_tokens, n_tokens // 4)
+            assert module(a, a, k_compression, v_compression)
+
+            # Linformer self-attention with the 'key-value' sharing policy
+            kv_compression = torch.nn.Linear(n_tokens, n_tokens // 4)
+            assert module(a, a, kv_compression, kv_compression)
+
+    References:
+        * [devlin2018bert] Jacob Devlin, Ming-Wei Chang, Kenton Lee, Kristina Toutanova "BERT: Pre-training of Deep Bidirectional Transformers for Language Understanding" 2018
+        * [wang2020linformer] Sinong Wang, Belinda Z. Li, Madian Khabsa, Han Fang, Hao Ma "Linformer: Self-Attention with Linear Complexity", 2020
+    """
+
+    def __init__(
+        self,
+        *,
+        d_token: int,
+        n_heads: int,
+        dropout: float,
+        bias: bool,
+        initialization: str,
+    ) -> None:
+        """
+        Args:
+            d_token: the token size. Must be a multiple of :code:`n_heads`.
+            n_heads: the number of heads. If greater than 1, then the module will have
+                an addition output layer (so called "mixing" layer).
+            dropout: dropout rate for the attention map. The dropout is applied to
+                *probabilities* and do not affect logits.
+            bias: if `True`, then input (and output, if presented) layers also have bias.
+                `True` is a reasonable default choice.
```

Comment:
[Torch by default](https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/activation.py#L905) has bias as False for k and v. It is using compiled [F.multi_head_attention_forward](https://github.com/pytorch/pytorch/blob/3d4136dc446068b802d52d5c925fa293df0e34cd/torch/nn/functional.py#L5063) call for performance.